File: chazm-model/src/main/java/runtimemodels/chazm/model/event/PmfEventFactory.java
package runtimemodels.chazm.model.event;

import runtimemodels.chazm.api.entity.Pmf;

/**
 * The {@linkplain PmfEventFactory} interface defines the API for constructing {@linkplain PmfEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface PmfEventFactory {

    /**
     * Constructs an {@linkplain PmfEvent}.
     *
     * @param category the {@linkplain EventType}.
     * @param pmf      the {@linkplain Pmf}.
     * @return a {@linkplain PmfEvent}.
     */
    PmfEvent build(EventType category, Pmf pmf);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/event/PmfEventFactory.kt
package runtimemodels.chazm.model.event

import runtimemodels.chazm.api.entity.Pmf

/**
 * The [PmfEventFactory] interface defines the API for constructing [PmfEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface PmfEventFactory {

    /**
     * Constructs an [PmfEvent].
     *
     * @param category the [EventType].
     * @param pmf      the [Pmf].
     * @return a [PmfEvent].
     */
    fun build(category: EventType, pmf: Pmf): PmfEvent

}
